### PR TITLE
Codemeta support

### PR DESCRIPTION
**Pull request by apirogov**
* Added codemeta support (using cffconvert + codemetapy)
* added new file to model and flags
* updated readme + removed complete cli argument dump

+ various other small changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -200,6 +198,7 @@ maintainers = [
 contributors = [
     {family-names = "Doe", given-names= "John", email = "test@test.test", orcid = "https://orcid.org/0000-0001-2345-5678", contribution = "The main author, maintainer and tester.", contribution_begin = "2023-03-01", contribution_type = "code"},
     {family-names = "Dow", given-names= "John", email = "test2@test.test", orcid = "https://orcid.org/0000-0012-3456-7890", contribution = "Reviewer", contribution_begin = "2023-03-01", contribution_type = "review"}
+    ]
```
Comment (mustafasoylu) [inline README.md:201]: there is already a ] in here

File: README.md [hunk 2]
```diff
@@ -232,6 +231,7 @@ maintainers = [
...
```
Comment (mustafasoylu) [inline README.md:234]: there is already a `]` in here

File: README.md (hunk 1 above)
Comment (apirogov) [reply README.md:201]: VS Code said it was not closed and there was no matching paren highlighted :man_shrugging:

File: src/somesy/cli/sync.py [hunk 3]
```diff
@@ -118,7 +138,9 @@ def sync(
 
         # check if there is at least one file to sync
         if prioritized_inputs.no_sync_pyproject and prioritized_inputs.no_sync_cff:
```
Comment (mustafasoylu) [inline src/somesy/cli/sync.py:140]: I think no-sync-codemeta should be in this if too

File: README.md (hunk 1 above)
Comment (apirogov) [reply README.md]: Okay no idea which brackets were wrong but I apparently fixed it now correctly :D

File: src/somesy/commands/sync.py [hunk 4]
```diff
@@ -3,35 +3,45 @@
...
     if pyproject_file is not None:
         _sync_python(metadata, pyproject_file)
 
     if cff_file is not None:
-        _sync_cff(metadata, cff_file, create_cff)
+        _sync_cff(metadata, cff_file)
+
+    # NOTE: codemeta should always be last, it uses (some of) the other targets
+    if codemeta_file is not None:
+        update_codemeta(codemeta_file, cff_file)
```
Comment (mustafasoylu) [inline src/somesy/commands/sync.py:44]: what happens if user sets no_cff_sync flag?

Comment (apirogov) [reply src/somesy/commands/sync.py:44]: Then none is passed in the calling method, exactly like with the other files.

File: src/somesy/cli/sync.py [hunk 5]
```diff
@@ -97,6 +115,8 @@ def sync(
             cff_file=cff_file,
             no_sync_pyproject=no_sync_pyproject,
             pyproject_file=pyproject_file,
+            no_sync_codemeta=no_sync_codemeta,
+            codemeta_file=codemeta_file,
```
Comment (mustafasoylu) [inline src/somesy/cli/sync.py:119]: COuld you update line 125-127 to print new cli inputs too?

**Review by mustafasoylu — COMMENTED** [review]
When I run somesy sync in this repo itself, it deletes my email in codemeta.json. Is it the expected behavior?

File: src/somesy/cli/sync.py (hunk 5 above)
Comment (apirogov) [reply src/somesy/cli/sync.py]: I was planning to... resulting in some serious refactoring :D but consider this done

**Comment by apirogov** [comment]
While refactoring, I also changed some stuff with the config logic.

1. The config has no "optional" fields. The config is complete. Optional fields are "partial" configurations (that we have no explicit model for). In the resulting config everything is non-optional
2. I moved validation logic into the model, where it belongs
3. I removed the "if path is set, change flag to false" thing, it was causing problems and I can see situations like you set the file in a config, but disable the sync in the cli, and this would lead to surprising behavior. I think setting the path and enabling sync are orthogonal things.

**Review by mustafasoylu — COMMENTED** [review]
when I run the command `somesy sync` codemeta.json changes as below 
```bash
diff --git a/codemeta.json b/codemeta.json
index 37a734c..d4675f5 100644
--- a/codemeta.json
+++ b/codemeta.json
@@ -11,17 +11,17 @@
     "audience": [
         {
             "@type": "Audience",
-            "audienceType": "Developers"
+            "audienceType": "Science/Research"
         },
         {
             "@type": "Audience",
-            "audienceType": "Science/Research"
+            "audienceType": "Developers"
         }
     ],
     "author": [
         {
+            "@id": "https://orcid.org/0000-0003-2637-0432",
             "@type": "Person",
-            "email": "m.soylu@fz-juelich.de",
             "familyName": "Soylu",
             "givenName": "Mustafa"
         }
```

**Comment by apirogov** [comment]
> when I run the command `somesy sync` codemeta.json changes as below
> 
> ```shell
> diff --git a/codemeta.json b/codemeta.json
> index 37a734c..d4675f5 100644
> --- a/codemeta.json
> +++ b/codemeta.json
> @@ -11,17 +11,17 @@
>      "audience": [
>          {
>              "@type": "Audience",
> -            "audienceType": "Developers"
> +            "audienceType": "Science/Research"
>          },
>          {
>              "@type": "Audience",
> -            "audienceType": "Science/Research"
> +            "audienceType": "Developers"
>          }
>      ],
>      "author": [
>          {
> +            "@id": "https://orcid.org/0000-0003-2637-0432",
>              "@type": "Person",
> -            "email": "m.soylu@fz-juelich.de",
>              "familyName": "Soylu",
>              "givenName": "Mustafa"
>          }
> ```

Yeah, because the graph changed (email out, orcid in). When you run it again, it will should update it again.

**Comment by mustafasoylu** [comment]
> Yeah, because the graph changed (email out, orcid in). When you run it again, it should not update it again.

Sorry, I don't understand it, Email is recorded in somesy config file, cff file and pyproject file so it should be in codemeta.json, right?

**Review by mustafasoylu — APPROVED** [review]
lgtm

**Pull request merged**